#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.1570424797675456 -0.8760052999629788 0.4560179535777419 5.5340754745876275
  position -1.4307557912261795 0.24295982201241204 1.1651853932201073
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
Robot {
  children [
    DEF body Transform {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.045 0 0.025
      }
      device [
        PositionSensor {
          name "pos1"
        }
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint Solid {
        translation 0.045 0 0.025
        rotation 0 1 7.853589927914978e-17 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.045 0 0.025
      }
      device [
        PositionSensor {
          name "pos2"
        }
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint Solid {
        translation -0.045 0 0.025
        rotation 0 1 0 1.57
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE wheel_physics
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "Odometry"
}
